### PR TITLE
Fix various bazel build issues on master

### PR DESCRIPTION
**Pull request by ZacSweers**
Identified following #1268

### CODE REVIEW

File: javatests/dagger/android/support/functional/BUILD [hunk 1]
```diff
@@ -33,6 +33,7 @@ android_library(
         "@androidsdk//com.android.support:appcompat-v7-25.0.0",
         "@androidsdk//com.android.support:support-fragment-25.0.0",
         "@google_bazel_common//third_party/java/guava",
+        "@javax_annotation_jsr250_api",
```
[the remark this answers is not in the document]
Comment (kirillzh) [reply javatests/dagger/android/support/functional/BUILD]: From #1268 
```
ERROR: /Users/zsweers/dev/java/contributing/dagger/javatests/dagger/android/support/functional/BUILD:20:1: Building javatests/dagger/android/support/functional/libfunctional.jar (13 source files) and running annotation processors (AndroidProcessor, ComponentProcessor) failed (Exit 1)
bazel-out/darwin-fastbuild/bin/javatests/dagger/android/support/functional/_javac/functional/libfunctional_sourcegenfiles/dagger/android/support/functional/UsesGeneratedModulesApplication_ApplicationModule_ContributeTestActivityWithScopeInjector.java:16: error: [strict] Using type javax.annotation.Generated from an indirect dependency (TOOL_INFO: "@javax_annotation_jsr250_api//:javax_annotation_jsr250_api"). See command below **
@Generated("dagger.android.processor.AndroidProcessor")
 ^
 ** Please add the following dependencies:
  @javax_annotation_jsr250_api//:javax_annotation_jsr250_api to //javatests/dagger/android/support/functional
 ** You can use the following buildozer command:
buildozer 'add deps @javax_annotation_jsr250_api//:javax_annotation_jsr250_api' //javatests/dagger/android/support/functional
```

File: .bazelrc [hunk 2]
```diff
@@ -6,3 +6,7 @@ build --host_javacopt=-g
 # TODO(ronshapiro): explore how much work it would be to reenable this
 build --javacopt="-Xep:BetaApi:OFF"
 build --host_javacopt="-Xep:BetaApi:OFF"
+
+# Legacy flags
+build --incompatible_package_name_is_a_function=false
```
[the remark this answers is not in the document]
Comment (kirillzh) [reply .bazelrc]: That PR is merged now, I attempted to use latest `basel-common` version but got:
```
ERROR: /private/var/tmp/_bazel_zhukov/ec789c8512b32a6fa14b8de7a8b5d17e/external/com_google_protobuf/BUILD:130:1: no such target '//external:zlib': target 'zlib' not declared in package 'external' defined by /Users/zhukov/Development/GitHub/google/dagger/WORKSPACE and referenced by '@com_google_protobuf//:protobuf'
ERROR: Analysis of target '//javatests/dagger/grpc/functional/server:coffee_service_java_proto' failed; build aborted: Analysis failed
INFO: Elapsed time: 2.174s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (191 packages loaded, 2385 targets configured)
FAILED: Build did NOT complete successfully (191 packages loaded, 2385 targets configured)
```

[the remark this answers is not in the document]
Comment (kirillzh) [reply .bazelrc]: Seems to be related to https://github.com/protocolbuffers/protobuf/issues/5472.

**Comment by ZacSweers** [comment]
Do you want me to revisit this?

**Comment by ronshapiro** [comment]
If there are still issues, I imagine those would be helpful. I'd recommend
trying to keep each change isolated if possible

בתאריך יום ג׳, 23 ביולי 2019, 22:35, מאת Zac Sweers ‏<
notifications@github.com>:

**Comment by ZacSweers** [comment]
`incompatible_disable_depset_in_cc_user_flags` is still an issue for me locally, the rest seem fine now. For android, I have to add `--experimental_google_legacy_api` now. Should I update this or make a new PR for each of those separately?

**Comment by ZacSweers** [comment]
Closing this out now as it's no longer relevant I think

**Pull request closed without merging**